### PR TITLE
Added code for Longest Common Subsequence in C# and Php

### PR DESCRIPTION
**Pull request by rbjoshi1309**
**Fixes issue:** #4871 

**Changes:**

Added C# and Php version for Longest Common Subsequence in following folder :
cosmos/code/dynamic_programming/src/longest_common_subsequence

### CODE REVIEW

File: code/dynamic_programming/src/longest_common_subsequence/longest_common_subsequence.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
+using System; 
+
+class LongestCommonSubsequence
+{ 
+    static int lcs(char[] x, char[] y, int x_len, int y_len)
```
Comment (arnavb) [inline code/dynamic_programming/src/longest_common_subsequence/longest_common_subsequence.cs]: Rename functions using upper camel case.

File: code/dynamic_programming/src/longest_common_subsequence/longest_common_subsequence.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+                    dp[i, j] = max(dp[i - 1, j], dp[i, j - 1]); 
+            } 
+        }
+
+        return dp[x_len, y_len]; 
+    } 
+
+    static int max(int a, int b) 
+    { 
+        return (a > b)? a : b;
```
Comment (arnavb) [inline code/dynamic_programming/src/longest_common_subsequence/longest_common_subsequence.cs]:
```suggestion
        return (a > b) ? a : b; 
```

File: code/dynamic_programming/src/longest_common_subsequence/longest_common_subsequence.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    } 
+
+    public static void Main() 
+    { 
+        String s1 = "AGGTAB"; 
+        String s2 = "GXTXAYB"; 
+
+        char[] a = s1.ToCharArray(); 
+        char[] b = s2.ToCharArray(); 
+        int a_len = a.Length;
```
Comment (arnavb) [inline code/dynamic_programming/src/longest_common_subsequence/longest_common_subsequence.cs]: These variables are not needed.

File: code/dynamic_programming/src/longest_common_subsequence/longest_common_subsequence.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+    public static void Main() 
+    { 
+        String s1 = "AGGTAB"; 
+        String s2 = "GXTXAYB"; 
+
+        char[] a = s1.ToCharArray(); 
+        char[] b = s2.ToCharArray(); 
+        int a_len = a.Length; 
+        int b_len = b.Length;
```
Comment (arnavb) [inline code/dynamic_programming/src/longest_common_subsequence/longest_common_subsequence.cs]: These variables are not needed.

File: code/dynamic_programming/src/longest_common_subsequence/longest_common_subsequence.php [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+        return $dp[$x_len][$y_len]; 
+    } 
+
+    $a = "AGGTAB"; 
+    $b = "GXTXAYB";
+
+    echo "Length of LCS : "; 
+    echo lcs($a, $b); 
+
+?>
```
Comment (arnavb) [inline code/dynamic_programming/src/longest_common_subsequence/longest_common_subsequence.php]: Don't have this.

File: code/dynamic_programming/src/longest_common_subsequence/longest_common_subsequence.php [hunk 6]
```diff
@@ -0,0 +1,30 @@
+<?php 
+
+    function lcs($x , $y)
```
Comment (arnavb) [inline code/dynamic_programming/src/longest_common_subsequence/longest_common_subsequence.php]: Dedent the entire file.

File: code/dynamic_programming/src/longest_common_subsequence/longest_common_subsequence.cs (hunk 1 above)
Comment (rbjoshi1309) [reply code/dynamic_programming/src/longest_common_subsequence/longest_common_subsequence.cs]: Done

File: code/dynamic_programming/src/longest_common_subsequence/longest_common_subsequence.cs (hunk 2 above)
Comment (rbjoshi1309) [reply code/dynamic_programming/src/longest_common_subsequence/longest_common_subsequence.cs]: Done

File: code/dynamic_programming/src/longest_common_subsequence/longest_common_subsequence.cs (hunk 4 above)
Comment (rbjoshi1309) [reply code/dynamic_programming/src/longest_common_subsequence/longest_common_subsequence.cs]: Done

File: code/dynamic_programming/src/longest_common_subsequence/longest_common_subsequence.php (hunk 5 above)
Comment (rbjoshi1309) [reply code/dynamic_programming/src/longest_common_subsequence/longest_common_subsequence.php]: Done

File: code/dynamic_programming/src/longest_common_subsequence/longest_common_subsequence.php (hunk 6 above)
Comment (rbjoshi1309) [reply code/dynamic_programming/src/longest_common_subsequence/longest_common_subsequence.php]: Done

**Comment by rbjoshi1309** [comment]
@arnavb Please have a look at this PR.

File: code/dynamic_programming/src/longest_common_subsequence/longest_common_subsequence.php (hunk 5 above)
Comment (arnavb) [reply code/dynamic_programming/src/longest_common_subsequence/longest_common_subsequence.php]: @rbjoshi1309 Still need to remove line 28. Also unindent the rest of the file.

**Comment by rbjoshi1309** [comment]
@arnavb Made all the required changes.

File: code/dynamic_programming/src/longest_common_subsequence/longest_common_subsequence.php (hunk 5 above)
Comment (rbjoshi1309) [reply code/dynamic_programming/src/longest_common_subsequence/longest_common_subsequence.php:27]: Done

File: code/dynamic_programming/src/longest_common_subsequence/longest_common_subsequence.php (hunk 6 above)
Comment (rbjoshi1309) [reply code/dynamic_programming/src/longest_common_subsequence/longest_common_subsequence.php]: @arnavb Done

File: code/dynamic_programming/src/longest_common_subsequence/longest_common_subsequence.php (hunk 5 above)
Comment (arnavb) [reply code/dynamic_programming/src/longest_common_subsequence/longest_common_subsequence.php:27]: @rbjoshi1309 You still haven't removed the trailing `?>`.

Comment (rbjoshi1309) [reply code/dynamic_programming/src/longest_common_subsequence/longest_common_subsequence.php]: @arnavb Done.

**Pull request merged**